flaml/model.py

```diff
@@ -2118,6 +2118,190 @@ class XGBoostLimitDepth_TS(TS_SKLearn):
+                "time_varying_known_categoricals", []
+            ),
+            time_varying_known_reals=kwargs.get("time_varying_known_reals", []),
+            time_varying_unknown_categoricals=kwargs.get(
+                "time_varying_unknown_categoricals", []
+            ),
+            time_varying_unknown_reals=kwargs.get("time_varying_unknown_reals", []),
+            variable_groups=kwargs.get(
+                "variable_groups", {}
+            ),  # group of categorical variables can be treated as one variable
+            lags=kwargs.get("lags", {}),
+            target_normalizer=GroupNormalizer(
+                groups=kwargs["group_ids"], transformation="softplus"
+            ),  # use softplus and normalize by group
+            add_relative_time_idx=True,
+            add_target_scales=True,
+            add_encoder_length=True,
+        )
+
+        # create validation set (predict=True) which means to predict the last max_prediction_length points in time
+        # for each series
+        validation = TimeSeriesDataSet.from_dataset(
+            training, self.data, predict=True, stop_randomization=True
+        )
+
+        # create dataloaders for model
+        batch_size = 128  # set this between 32 to 128
+        train_dataloader = training.to_dataloader(
+            train=True, batch_size=batch_size, num_workers=0
+        )
+        val_dataloader = validation.to_dataloader(
+            train=False, batch_size=batch_size * 10, num_workers=0
+        )
+
+        return training, train_dataloader, val_dataloader
+
+    def fit(self, X_train, y_train, budget=None, **kwargs):
+        import copy
+        from pathlib import Path
+        import warnings
+        import numpy as np
+        import pandas as pd
+        import pytorch_lightning as pl
+        from pytorch_lightning.callbacks import EarlyStopping, LearningRateMonitor
+        from pytorch_lightning.loggers import TensorBoardLogger
+        import torch
+        from pytorch_forecasting import TemporalFusionTransformer
+        from pytorch_forecasting.metrics import QuantileLoss
+        import tensorflow as tf
+        import tensorboard as tb
+
+        warnings.filterwarnings("ignore")
+        tf.io.gfile = tb.compat.tensorflow_stub.io.gfile
+        current_time = time.time()
+        training, train_dataloader, val_dataloader = self.transform_ds(
+            X_train, y_train, **kwargs
+        )
+        params = self.params.copy()
+        gradient_clip_val = params.pop("gradient_clip_val")
+        params.pop("n_jobs")
+        max_epochs = 1
+        early_stop_callback = EarlyStopping(
+            monitor="val_loss", min_delta=1e-4, patience=10, verbose=False, mode="min"
+        )
+        lr_logger = LearningRateMonitor()  # log the learning rate
+        logger = TensorBoardLogger("lightning_logs")  # logging results to a tensorboard
+        default_trainer_kwargs = dict(
+            gpus=[0] if torch.cuda.is_available() else None,

```

Comment:
why restricting to one gpu? Shall we alllow `gpu_per_trial` like TransformersEstimator?